lib/bureaucrat/swagger_slate_markdown_writer.ex

```diff
@@ -0,0 +1,328 @@
+  end
+
+  @doc """
+  Writes all examples of a given operation (Controller action) to file.
+  """
+  def write_action(file, operation_id, records, swagger) do
+    details = find_operation_by_id(swagger, operation_id)
+    puts(file, "## #{details["summary"]}\n")
+
+    # write examples before params/schemas to get correct alignment in slate
+    Enum.each records, fn record ->
+      write_example(file, record)
+    end
+
+    file
+    |> puts("#{details["description"]}\n")
+    |> write_parameters(details)
+    |> write_responses(details)
+  end
+
+  @doc """
+  Find the details of an API operation in swagger by operationId
+  """
+  def find_operation_by_id(swagger, operation_id) do
+    Enum.flat_map(swagger["paths"], fn {_path, actions} ->
+      Enum.map(actions, fn {_action, details} -> details end)
+    end)
+    |> Enum.find(fn details ->
+      details["operationId"] == operation_id
+    end)
+  end
+
+  @doc """
+  Writes the parameters table for given swagger operation to file.
+
+  Uses the vendor extension "x-example" to provide example of each parameter.
+  TODO: detailed schema validation rules aren't shown yet (min/max/regex/etc...)
+  """
+  def write_parameters(file, swagger_operation) do
+    file
+    |> puts("#### Parameters\n")
+    |> puts("| Parameter   | Description | In |Type      | Required | Default | Example |")
+    |> puts("|-------------|-------------|----|----------|----------|---------|---------|")
+
+    Enum.each swagger_operation["parameters"], fn param ->
+      content =
+        ["name", "description", "in", "type", "required", "default", "x-example"]
+        |> Enum.map(&(param[&1]))
+        |> Enum.join("|")
+      puts(file, "|#{content}|")
+    end
+    puts file, ""
+  end
+
+  @doc """
+  Writes the responses table for given swagger operation to file.
+
+  Swagger only allows a single description per status code, which can be limiting
+   when trying to describe all possible error responses.  To work around this, add
+   markdown links into the description.
+  """
+  def write_responses(file, swagger_operation) do
+    file
+    |> puts("#### Responses\n")
+    |> puts("| Status | Description | Schema |")
+    |> puts("|--------|-------------|--------|")
+
+    Enum.each swagger_operation["responses"], fn {status, response} ->
+      type = response["schema"]["$ref"] |> String.replace_prefix("#/definitions/", "")
+      puts(file, "|#{status} | #{response["description"]} | [#{type}](##{String.downcase(type)})|")
+    end
+  end
+
+  @doc """
+  Writes a single request/response example to file
+  """
+  def write_example(file, record) do
+    path = case record.query_string do
+      "" -> record.request_path
+      str -> "#{record.request_path}?#{str}"
+    end
+
+    # Request with path and headers
+    file
+    |> puts("> #{record.assigns.bureaucrat_desc}\n")
+    |> puts("```plaintext")
+    |> puts("#{record.method} #{path}")
+    Enum.each record.req_headers, fn {header, value} ->
```

Comment:
i prefere refactor it, maybe with recursion? 
and for empty `[]` use 
`puts(file, "```\n")`